prov/util/src/util_mem_hooks.c

```diff
@@ -283,28 +283,22 @@ static void ofi_restore_intercepts(void)
 {
 	struct ofi_intercept *intercept;
 
-	fastlock_acquire(&memhooks_monitor->lock);
 	dlist_foreach_container(&memhooks.intercept_list, struct ofi_intercept,
 		intercept, entry) {
 		dl_iterate_phdr(ofi_restore_phdr_handler, intercept);
 	}
-	fastlock_release(&memhooks_monitor->lock);
 }
 
 static int ofi_intercept_symbol(struct ofi_intercept *intercept, void **real_func)
 {
 	int ret;
 
-	/*
-	* Take lock first to handle a possible race where dlopen() is called
-	* from another thread and we may end up not patching it.
-	*/
 	FI_DBG(&core_prov, FI_LOG_MR,
 	       "intercepting symbol %s\n", intercept->symbol);
-	fastlock_acquire(&memhooks_monitor->lock);
+	dlist_init(&intercept->dl_intercept_list);
```

Comment:
Can you move this to ofi_memhooks_init()?  We can iterate over the intercept array there.  It may be easier to add OFI_INTERCEPT_MAX to the enum for this purpose.

Reply:
@shefty are we talking about something like this?
```
for (i = 0; i < OFI_INTERCEPT_MAX; ++i) {
        dlist_init(&intercepts[i]->dl_intercept_list);
}
```

Reply:
Yes - that would be perfect

Reply:
Done!